src/groups/mqb/mqbnet/mqbnet_tcpsessionfactory.cpp

```diff
@@ -1414,5 +1409,34 @@ bool TCPSessionFactory::isEndpointLoopback(const bslstl::StringRef& uri) const
            mwcio::ChannelUtil::isLocalHost(endpoint.host());
 }
 
+// --------------------------------------
+// class TCPSessionFactory::PortExtractor
+// --------------------------------------
+
+TCPSessionFactory::PortExtractor::PortExtractor(bslma::Allocator* allocator)
+: d_regex(allocator)
+{
+    const char                  pattern[] = ":(\\d{1,5})$";
+    bsl::string                 error(allocator);
+    size_t                      errorOffset;
+    BSLA_MAYBE_UNUSED const int rc = d_regex.prepare(
+        &error,
+        &errorOffset,
+        pattern,
+        bdlpcre::RegEx::k_FLAG_JIT);
+    BSLS_ASSERT_SAFE(rc == 0);
+    BSLS_ASSERT_SAFE(d_regex.isPrepared() == true);
+}
+
+bsl::string_view
+TCPSessionFactory::PortExtractor::extract(const bsl::string& endpoint) const
+{
+    bsl::string_view result;
+
+    BSLS_ASSERT_SAFE(0 == d_regex.match(&result, endpoint));
```

Comment:
Is it possible to pass an incorrect endpoint here and cause the broker to crash on assert? I am thinking if we should crash here or handle incorrect `endpoint`

Reply:
It never happens as an endpoint is always a `host:port` string. However, I can remove this assert and just return an empty string in case of not matching endpoint. Then broker will proceed working and all the channels with unknown ports will report their TCP connections numbers metric with an empty tag port